lib/MessageBuilder.ts

```diff
@@ -0,0 +1,151 @@
+import * as zod from 'zod';
+import { type NonEmptyArray, isNonEmptyArray } from './utils/NonEmptyArray.ts';
+import { joinPath } from './utils/joinPath.ts';
+import {
+  ISSUE_SEPARATOR,
+  MAX_ISSUES_IN_MESSAGE,
+  PREFIX,
+  PREFIX_SEPARATOR,
+  UNION_SEPARATOR,
+} from './config.ts';
+
+export type ZodIssue = zod.ZodIssue;
+
+export type MessageBuilder = (issues: NonEmptyArray<ZodIssue>) => string;
```

Comment:
[suggestion] Related to my previous comment

```suggestion
export type MessageBuilder = (input: ZodIssue | NonEmptyArray<ZodIssue>) => string;
```

Reply:
But that would complicate things for devs who want to create their own version of `MessageBuilder`. They would need to cover for both `ZodIssue` and well as `NonEmptyArray<ZodIssue>`.

Reply:
I don't think is much of an overhead tbh, but I see your point. Resolving

Reply:
I don't have a strong opinion here. If you think this isn't much of an issue, then sure - np.

Reply:
Nevermind, let's keep it as is. At the end of the day, only the lib itself from an internal util we'll use the scalar issue param.